java/cdk-environment/src/main/java/sleeper/environment/cdk/networking/NetworkingStack.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright 2022 Crown Copyright
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package sleeper.environment.cdk.networking;
+
+import software.amazon.awscdk.NestedStack;
+import software.amazon.awscdk.services.ec2.*;
+import software.constructs.Construct;
+
+import java.util.Arrays;
+import java.util.Collections;
+
+public class NetworkingStack extends NestedStack {
+
+    private final Vpc vpc;
+
+    public NetworkingStack(Construct scope) {
+        super(scope, "Networking");
+
+        vpc = Vpc.Builder.create(this, "Vpc")
+                .cidr("10.150.0.0/16")
+                .maxAzs(1)
+                .subnetConfiguration(Arrays.asList(
+                        SubnetConfiguration.builder().name("public")
+                                .subnetType(SubnetType.PUBLIC)
+                                .cidrMask(24).build(),
+                        SubnetConfiguration.builder().name("private")
+                                .subnetType(SubnetType.PRIVATE_WITH_NAT)
+                                .cidrMask(24).build()))
+                .build();
```

Comment:
Similar to the comment about the key name, can we let the user provide an id here that will be used as part of the VPC name? Currently the name is "SleeperEnvironment/Networking/Vpc" - we might get a clash if two people deploy the stack to the same account?


<br class="Apple-interchange-newline" style="caret-color: rgb(0, 0, 0); color: rgb(0, 0, 0); font-style: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: auto; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: auto; word-spacing: 0px; -webkit-text-size-adjust: auto; -webkit-text-stroke-width: 0px; text-decoration: none;">

Reply:
Done